pkg/fs/dir.go

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2019 The virnsnap authors. See file "AUTHORS".
+// Licensed under the MIT License. You have obtained a copy of the License at
+// the "LICENSE" file in this repository.
+
+// Package fs implements helper functions for handling filesystem related
+// tasks.
+package fs
+
+import (
+	"fmt"
+	"os"
+
+	"golang.org/x/sys/unix"
+)
+
+// EnsureDirectory is a helper function that ensures path points to a directory.
+// If the directory does not exist, create it. The directory needs to be
+// accessible and writeable so the check can pass.
+func EnsureDirectory(path string) error {
+	stat, err := os.Stat(path)
```

Comment:
The below block is very verbose and I think it could be simplified with using [`os.MkdirAll()`](https://golang.org/pkg/os/#MkdirAll), which will either create the directory or do nothing if it already exists

Reply:
You're right. I kind of reimplemented this function. I quickly checked its [source code](https://golang.org/src/os/path.go?s=497:544#L8) and removed the EnsureDirectory function and its calls completely.